### PR TITLE
check-browser-compatibility-kartoza/geosafe#64

### PR DESCRIPTION
**Pull request by boney-bun**
check a browser compatibility.

check if a browser comply with ECMA Script 5 and able to load JSON file.
show a pop up window if it doesn't.
implement kartoza/geosafe#64.

### CODE REVIEW

**Comment by boney-bun** [comment]
@lucernae @gubuntu please review the code

**Comment by lucernae** [comment]
@boney-bun please change PR title and commit message to something more descriptive next time.

File: geonode/templates/base.html [hunk 1]
```diff
@@ -5,28 +5,21 @@
...
-    <meta Http-Equiv="Cache-directive: no-cache">
-
     <title>{% block title %}{{ SITE_NAME }}{% endblock %}</title>
     {% block head %}
       {% if DEBUG_STATIC %}
-          <link href="{{ STATIC_URL }}lib/css/jquery.dataTables.css?v={{ VERSION }}" rel="stylesheet" />
-          <link href="{{ STATIC_URL }}lib/css/select2.css?v={{ VERSION }}" rel="stylesheet"/>
-          <link href="{{ STATIC_URL }}lib/css/bootstrap.min.css?v={{ VERSION }}" rel="stylesheet"/>
-          <link href="{{ STATIC_URL }}lib/css/jquery-ui.css?v={{ VERSION }}" rel="stylesheet"/>
+      <link href="{{ STATIC_URL }}lib/css/jquery.dataTables.css?v={{ VERSION }}" rel="stylesheet" />
```
Comment (lucernae) [inline geonode/templates/base.html]: indent in this block

File: geonode/templates/base.html [hunk 2]
```diff
@@ -5,28 +5,21 @@
...
+      <link href="{{ STATIC_URL }}lib/css/select2.css?v={{ VERSION }}" rel="stylesheet"/>
+      <link href="{{ STATIC_URL }}lib/css/bootstrap.min.css?v={{ VERSION }}" rel="stylesheet"/>
+      <link href="{{ STATIC_URL }}lib/css/jquery-ui.css?v={{ VERSION }}" rel="stylesheet"/>
+      <link href="{{ STATIC_URL }}lib/css/bootstrap-datetimepicker.css?v={{ VERSION }}" rel="stylesheet"/>
+      <link href="{{ STATIC_URL }}lib/css/bootstrap-tokenfield.css?v={{ VERSION }}" rel="stylesheet"/>
       {% else %}
-        <link href="{{ STATIC_URL }}lib/css/assets.min.css?v={{ VERSION }}" rel="stylesheet"/>
+      <link href="{{ STATIC_URL }}lib/css/assets.min.css?v={{ VERSION }}" rel="stylesheet"/>
```
Comment (lucernae) [inline geonode/templates/base.html]: Indent this too, the same level with above

File: geonode/templates/base.html [hunk 3]
```diff
@@ -151,6 +144,28 @@
       <p>Please use a <strong>modern browser</strong> like Mozilla Firefox, Google Chrome or Safari.</p>
     </div>
```
Comment (lucernae) [inline geonode/templates/base.html]: This is also an alert for outdated browser. So, it does exists in original code. We have to make sure there is no conflict of interests. There might be existing methods for Geonode to check outdated browser.

File: geonode/templates/base.html [hunk 4]
```diff
@@ -5,28 +5,21 @@
...
-          <link href="{{ STATIC_URL }}lib/css/jquery.dataTables.css?v={{ VERSION }}" rel="stylesheet" />
-          <link href="{{ STATIC_URL }}lib/css/select2.css?v={{ VERSION }}" rel="stylesheet"/>
-          <link href="{{ STATIC_URL }}lib/css/bootstrap.min.css?v={{ VERSION }}" rel="stylesheet"/>
-          <link href="{{ STATIC_URL }}lib/css/jquery-ui.css?v={{ VERSION }}" rel="stylesheet"/>
+      <link href="{{ STATIC_URL }}lib/css/jquery.dataTables.css?v={{ VERSION }}" rel="stylesheet" />
+      <link href="{{ STATIC_URL }}lib/css/select2.css?v={{ VERSION }}" rel="stylesheet"/>
+      <link href="{{ STATIC_URL }}lib/css/bootstrap.min.css?v={{ VERSION }}" rel="stylesheet"/>
+      <link href="{{ STATIC_URL }}lib/css/jquery-ui.css?v={{ VERSION }}" rel="stylesheet"/>
+      <link href="{{ STATIC_URL }}lib/css/bootstrap-datetimepicker.css?v={{ VERSION }}" rel="stylesheet"/>
+      <link href="{{ STATIC_URL }}lib/css/bootstrap-tokenfield.css?v={{ VERSION }}" rel="stylesheet"/>
```
Comment (lucernae) [inline geonode/templates/base.html]: Forgot to review this.
You added datetimepicker and tokenfield.
Is this related with the library?
Because I didn't see this file in static folder. How did you add this?

File: geonode/templates/base.html (hunk 3 above)
Comment (boney-bun) [inline geonode/templates/base.html]: I have examined this code. I think it specifically targets at the old IE browser. The code and explanation is [here](https://github.com/afarkas/html5shiv)

File: geonode/templates/base.html (hunk 4 above)
Comment (boney-bun) [inline geonode/templates/base.html]: I didn't add this library @lucernae 
Are you sure it was me?

Comment (lucernae) [inline geonode/templates/base.html]: Of course I'm sure, because it was coming from your diff :D.
+ line means, it was a line that you added.

File: geonode/templates/base.html (hunk 3 above)
Comment (lucernae) [inline geonode/templates/base.html]: What I meant is, you should probably ask Geonode developers for review wether to improve existing codes or change it. Because I don't know what's best for these and perhaps they have their own reasons/standard to implement this.

**Comment by boney-bun** [comment]
I close this PR and make another one.
I working on implementing the browser compatibility in geosafe, instead of geonode.
